### PR TITLE
Added unit tests in athena-db2.

### PR DESCRIPTION
**Pull request by ritiktrianz**
*Issue #, if available:*

*Description of changes:*
Added unit tests for athena-db2 connector.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-db2/src/test/java/com/amazonaws/athena/connectors/db2/Db2FederationExpressionParserTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.amazonaws.athena.connectors.db2.Db2Constants.QUOTE_CHARACTER;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class Db2FederationExpressionParserTest
+{
+    private static final String VALUE_1 = "value1";
```
Comment (ejeffrli) [inline athena-db2/src/test/java/com/amazonaws/athena/connectors/db2/Db2FederationExpressionParserTest.java]: think this constant is not needed.

Comment (Jithendar12) [reply athena-db2/src/test/java/com/amazonaws/athena/connectors/db2/Db2FederationExpressionParserTest.java]: Hi @ejeffrli, removed the VALUE_1 constant and updated all references to use string literals. Thank you!

**Review by hareeshkumar-t — DISMISSED** [review]

**Review by fal-bharadwaj — DISMISSED** [review]

**Comment by fal-bharadwaj** [comment]
@ritiktrianz @Jithendar12 looks like the build failed, please take a look, thanks.

**Comment by ritiktrianz** [comment]
> @ritiktrianz @Jithendar12 looks like the build failed, please take a look, thanks.

Hi @fal-bharadwaj ,
The build failure has been resolved. The issue originated from the recent merge, where a test invoked Assert.assertEquals but the import for the Assert class had been removed in this PR. This has now been fixed. Thank you!

**Review by fal-bharadwaj — DISMISSED** [review]

**Review by burhan94 — COMMENTED** [review]
## Code Review

### ✅ Looks Good

Good test coverage for DB2 connector including query passthrough and edge cases. Proper @After cleanup.

### 🟢 Minor suggestions
- Consider avoiding RETURNS_DEEP_STUBS where possible.
- Some test methods exceed 25 lines — could be split.

**Comment by Jithendar12** [comment]
> ## Code Review
> ### ✅ Looks Good
> 
> Good test coverage for DB2 connector including query passthrough and edge cases. Proper @after cleanup.
> ### 🟢 Minor suggestions
> 
>     * Consider avoiding RETURNS_DEEP_STUBS where possible.
> 
>     * Some test methods exceed 25 lines — could be split.

@burhan94, replaced RETURNS_DEEP_STUBS with explicit mocks.

**Review by hareeshkumar-t — APPROVED** [review]

**Review by AbdulR3hman — APPROVED** [review]
